server/src/main/java/org/elasticsearch/common/geo/builders/PolygonBuilder.java

```diff
@@ -678,6 +672,30 @@ private static int createEdges(int component, Orientation orientation, LineStrin
         return concat(component, direction ^ orientation, points, offset, edges, toffset, length);
     }
 
+    /**
+     * @return whether the points are clockwise (true) or anticlockwise (false)
+     */
+    private static boolean getOrientation(Coordinate[] points, int offset, int length) {
+        // calculate the direction of the points: find the southernmost point
+        // and check its neighbors orientation.
+
+        final int top = top(points, offset, length);
+        final int prev = (top + length - 1) % length;
+        final int next = (top + 1) % length;
+
+        // This duplicates the functionality of org.apache.lucene.geo.Polygon2D#orient which is, at time of writing, private.
+        // TODO make Lucene's method public and then use it here instead.
+        final double determinant
+            = (points[offset + next].x - points[offset + top].x) * (points[offset + prev].y - points[offset + top].y)
+            - (points[offset + prev].x - points[offset + top].x) * (points[offset + next].y - points[offset + top].y);
+        assert determinant != 0.0;
```

Comment:
I have a couple questions about this assertion:
- I think the determinant could be 0 if the three points lie in a straight line. Is my understanding correct that this won't happen because of the way we choose `top`? It might be nice to add a quick note to explain.
- Are we concerned about numeric underflow here? I'm curious as to how we handle numeric issues in other parts of the geo code (in particular, whether we should throw a runtime exception when an issue is detected).